packages/engine/src/avatar/components/TransparencyDitheringComponent.tsx

```diff
@@ -76,36 +81,51 @@ export const TransparencyDitheringComponent = Array.from({ length: maxDitherPoin
     }
   })
 })
-export const TransparencyDitheringPlugin: PluginObjectType = {
-  id: 'TransparencyDithering',
-  priority: 10,
-  compile: (shader, renderer) => {
-    const pluginEntity = pluginByName[TransparencyDitheringPlugin.id]
-    const plugin = getOptionalMutableComponent(pluginEntity, MaterialComponent[MaterialComponents.Plugin])
-    if (!plugin) return
-
-    if (!shader.vertexShader.startsWith('varying vec3 vWorldPosition')) {
-      shader.vertexShader = shader.vertexShader.replace(
-        /#include <common>/,
-        '#include <common>\n' + ditheringVertexUniform
-      )
+
+export const TransparencyDitheringPlugin = defineComponent({
+  name: 'TransparencyDithering',
+  onInit: (entity) => {
+    return {
+      centers: new Uniform(Array.from({ length: maxDitherPoints }, () => new Vector3())),
+      exponents: new Uniform(Array.from({ length: maxDitherPoints }, () => 2)),
+      distances: new Uniform(Array.from({ length: maxDitherPoints }, () => 3)),
+      useWorldCalculation: new Uniform(Array.from({ length: maxDitherPoints }, () => false))
     }
-    shader.vertexShader = shader.vertexShader.replace(
-      /#include <worldpos_vertex>/,
-      '	#include <worldpos_vertex>\n' + ditheringVertex
-    )
-    if (!shader.fragmentShader.startsWith('varying vec3 vWorldPosition'))
-      shader.fragmentShader = shader.fragmentShader.replace(
-        /#include <common>/,
-        '#include <common>\n' + ditheringFragUniform
+  },
+
+  reactor: () => {
+    const entity = useEntityContext()
+    const materialComponent = getComponent(entity, MaterialComponent[MaterialComponents.State])
+    const material = materialComponent.material as Material
+    addOBCPlugin(material, (shader) => {
```

Comment:
this should be a in a useEffect

Reply:
that is the plan - this is unfinished and i basically just pasted it in a reactor forgetting to put it in a useeffect callback